dd-sdk-android/src/test/kotlin/com/datadog/android/utils/DatadogExt.kt

```diff
@@ -53,7 +53,15 @@ inline fun <reified T : Context> mockContext(
  */
 fun resolveTagName(caller: Any, defaultIfNotDebug: String? = null): String {
     return if (Datadog.isDebug) {
-        caller.javaClass.simpleName
+        val javaClass = caller.javaClass
+        val strippedName = javaClass.simpleName.replace(Regex("(\\$\\d+)+$"), "")
```

Comment:
This is the RegEx for the Anonymous class, are you sure you want to use this here ?

Reply:
Well, yes, this is so that tags from the tests match the tags generated in the lib